Services: Intrusion Detection - refactor pcap/netmap selection to "Capture mode" and add new "divert" option.

AdSchellevis opened the pull request:
With divert we can integrate suricata in firewall rules, which makes it easier to bypass large flows. This change requires the new SO_REUSEPORT_LB option in the kernel in order to distribute traffic over multiple workers.

AdSchellevis committed:
Services: Intrusion Detection - refactor pcap/netmap selection to "Capture mode" and add new "divert" option.

With divert we can integrate suricata in firewall rules, which makes it easier to bypass large flows.
This change requires the new SO_REUSEPORT_LB option in the kernel in order to distribute traffic over multiple workers.

fichtner commented on src/opnsense/mvc/app/controllers/OPNsense/IDS/forms/generalSettings.xml:
```diff
@@ -10,23 +10,32 @@
+        <label>Listeners</label>
+        <style>ids_mode ids_mode_divert</style>
+        <type>text</type>
+        <help>Specify the number of listeners to initiate, usually this equals the number of cpus in your system.</help>
```
```suggestion
        <help>Specify the number of listeners to initiate, usually this equals the number of CPUs in your system.</help>
```

fichtner commented on src/opnsense/mvc/app/models/OPNsense/IDS/Migrations/M1_1_2.php:
```diff
@@ -0,0 +1,46 @@
+    public function run($model)
+    {
+        $cnf = Config::getInstance()->object();
+        if ($cnf?->OPNsense?->IDS?->general?->ips == '1') {
```
no cast here?

AdSchellevis replied:
should be safe, it's a config (simplexml) object.

AdSchellevis committed:
Update src/opnsense/mvc/app/controllers/OPNsense/IDS/forms/generalSettings.xml

Co-authored-by: Franco Fichtner <franco@opnsense.org>

AdSchellevis committed:
Update src/opnsense/mvc/app/controllers/OPNsense/IDS/forms/generalSettings.xml

Co-authored-by: Franco Fichtner <franco@opnsense.org>

AdSchellevis committed:
Services: Intrusion Detection - hook "divert-to" into our new firewall ui.

Currently we only support a single divert-to target, but if we would like to integrate in the future with other services, it's practical to be able to offer a list of them,
list_divert_sockets.php acts as a stepping stone for this purpose, which now just returns a static list of one (8000 -> idps).

AdSchellevis committed:
Firewall: Rules [new] - add temporary validation to only support divert-to on IPv4 (as long as IPv6 doesn't work properly in pf)

fichtner commented:
🥳
